### PR TITLE
Bugfix: Add file_token field to dag schema

### LINKED ISSUE #12121: Bugs in REST API get dag details endpoint.
Here are some bugs in dag details endpoint.

1. Calling this endpoint `dags/{dag_id}/details` returns an error due to `doc_md` field that's not nullable.

2. The dag details endpoint `GET /dags/{dag_id}/details` does not return a json if dag does not exist.

3. This endpoint does not return a `file_token` which is needed to get the source code of a DAG. And also expected judging from the documentation.

**Apache Airflow version**: 2.0

**What happened**:
1. Calling this endpoint GET /dags/{dag_id}/details returns an error
2. When the above is fixed and you look for the details of a dag that does not exist, this endpoint returns a webpage instead of json.
3. Calling the endpoint with a dag ID that exists doesn't return a file_token

**What you expected to happen**:
1. I expected it to return the details of the dag
2. I expected it to return a json indicating that the dag was not found if dag doesn't exist
3. I expected it to return a file_token in the list of fields when the dag exists

**How to reproduce it**:
Run airflow in breeze.
Make a request to `/dags/{dag_id}/details`. It will return an error.
Go to openapi.yaml and set `doc_md` to be nullable.
Make another request to see that it returns a result without a file_token.
Change the dag id to a dag ID that does not exist.
Make another request and see that it returns a webpage.

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Closes: #12121 
This PR adds file_token to dag schema so we can be able to get dag source through API. Resolves the last problem(no 3) in #12121

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/api_connexion/schemas/dag_schema.py [hunk 1]
```diff
@@ -17,10 +17,12 @@
 
 from typing import List, NamedTuple
 
+from itsdangerous import URLSafeSerializer
```
Comment (kaxil) [inline airflow/api_connexion/schemas/dag_schema.py:20]: Do we need to add this as a dependency in setup.py?

Comment (ashb) [reply airflow/api_connexion/schemas/dag_schema.py:20]: We should, yes. It'll already be pulled in from flask/jinja, but we should record we use it directly.

File: tests/api_connexion/endpoints/test_dag_endpoint.py [hunk 2]
```diff
@@ -110,22 +115,21 @@ def test_should_respond_200(self):
         self._create_dag_models(1)
         response = self.client.get("/api/v1/dags/TEST_DAG_1", environ_overrides={'REMOTE_USER': "test"})
         assert response.status_code == 200
-
-        current_response = response.json
-        current_response["fileloc"] = "/tmp/test-dag.py"
+        file_token = SERIALIZER.dumps("/tmp/dag_1.py")
```
Comment (ashb) [inline tests/api_connexion/endpoints/test_dag_endpoint.py]: This is going to be constant as long as the secret key is constant, right? (And that is defined in our test config). So could we include the actual result once in one test, to ensure that it is actually encoding it correctly.

Comment (ephraimbuddy) [reply tests/api_connexion/endpoints/test_dag_endpoint.py]: Ok. Working on it

File: airflow/api_connexion/schemas/dag_schema.py (hunk 1 above)
Comment (ephraimbuddy) [reply airflow/api_connexion/schemas/dag_schema.py:20]: Should I add it as flask dependency e.g flask=['itsdangerous'] in setup.py?

Comment (ashb) [reply airflow/api_connexion/schemas/dag_schema.py:20]: Add it directly in to this list https://github.com/apache/airflow/blob/master/setup.cfg#L81-L134

**Review by mik-laj — APPROVED** [review]

**Review by ashb — APPROVED** [review]

**Pull request merged**